react-components/src/components/Reveal3DResources/ResourceUpdater.ts

```diff
@@ -0,0 +1,330 @@
+
+    await this._image360StylingHandler.setCollections(allModels);
+    allModels.forEach(async (model) => {
+      await this.applyStylingAndTransform(model);
+    });
+    return allModels;
+  }
+
+  private removeResources(
+    resources: AddResourceOptions[],
+    currentModels: AddOptionsWithModel[]
+  ): AddOptionsWithModel[] {
+    const currentModelsSet = new Set(currentModels);
+    resources.forEach((resourceAddOptions) => {
+      const resourceAndModel = [...currentModelsSet].find((model) =>
+        isSameModel(model.addOptions, resourceAddOptions)
+      );
+
+      assert(resourceAndModel !== undefined);
+
+      currentModelsSet.delete(resourceAndModel);
+
+      if (is3dModelOptions(resourceAddOptions)) {
+        this._viewer.removeModel(resourceAndModel.model as CogniteModel);
+      } else if (is360ImageCollectionOptions(resourceAddOptions)) {
+        this._viewer.remove360ImageSet(resourceAndModel.model as Image360Collection);
+      } else {
+        throw Error('Tried to remove Model of unknown type');
+      }
+    });
+
+    return [...currentModelsSet];
+  }
+
+  private addResources(resources: AddResourceOptions[]): Array<Promise<AddOptionsWithModel>> {
+    return resources.map(async (addOptions) => {
+      try {
+        const addedModel = await (async () => {
+          if (is3dModelOptions(addOptions)) {
+            const addedModel = await this._viewer.addModel(addOptions);
+            if (addedModel instanceof CogniteCadModel) {
+              addedModel.setDefaultNodeAppearance({ visible: false });
+              return { type: 'cad' as const, model: addedModel, addOptions };
+            } else {
+              addedModel.setDefaultPointCloudAppearance({ visible: false });
+              return { type: 'pointcloud' as const, model: addedModel, addOptions };
+            }
+          } else {
+            const addedCollection = await (async () => {
+              if (is360DataModelCollection(addOptions)) {
+                return await this._viewer.add360ImageSet('datamodels', {
+                  image360CollectionExternalId: addOptions.externalId,
+                  space: addOptions.space
+                });
+              } else {
+                return await this._viewer.add360ImageSet('events', {
+                  site_id: addOptions.siteId
+                });
+              }
+            })();
+
+            addedCollection.setDefaultAnnotationStyle({ visible: false });
+            return {
+              model: addedCollection,
+              addOptions,
+              type: 'image360' as const
+            };
+          }
+        })();
+
+        this._onModelLoaded?.();
+        return addedModel;
+      } catch (error: any) {
+        this._onModelLoadedError?.(addOptions, error);
+        throw error;
+      }
+    });
+  }
+
+  private async applyStylingAndTransform(model: AddOptionsWithModel): Promise<void> {
+    if (model.type === 'cad') {
```

Comment:
Can we combine model exists and model type check in one `if` loop and `return`early and also can we use `switch` in place of `ìf else` statement, it will be more organized and shorten the code 🤔

Reply:
Good point, did a refactor :+1: